runner.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+type runner struct {
+	configuration      *configuration
+	backend            backend
+	cancelChan         chan bool
+	respawnWorkerDelay time.Duration
+	respawnWorkerChan  chan *rule
+}
+
+func (rn *runner) initialize() error {
```

Comment:
Where are ```cancelChan```, ```respawnWorkerDelay``` and ```respawnWorkerChan``` initialised?

Reply:
newRunner. If the type was exported, you could think about "making the zero value work" (which usually entails intialization in many places and unnecessary nil checks...).